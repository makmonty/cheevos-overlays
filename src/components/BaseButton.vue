<script lang="ts" setup>
const {type = 'button'} = defineProps<{
	type: 'button' | 'submit'
	disabled?: boolean
}>()
</script>

<template>
	<button :type="type" class="button" :disabled="disabled">
		<slot />
	</button>
</template>

<style scoped>
.button {
	border: 1px solid transparent;
	background: var(--info-color);
	border-radius: var(--default-border-radius);
}
</style>
