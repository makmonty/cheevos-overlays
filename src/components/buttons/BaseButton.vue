<script lang="ts" setup>
const {
  type = 'button',
  variant = 'button',
  size = 'medium'
} = defineProps<{
  type?: 'button' | 'submit';
  variant?: 'button' | 'text';
  size?: 'small' | 'medium' | 'large';
  disabled?: boolean;
}>();

const emit = defineEmits<{
  (e: 'click'): void;
}>();
</script>

<template>
  <button
    :type="type"
    class="button"
    :class="{ [`variant-${variant}`]: true, [`size-${size}`]: true }"
    :disabled="disabled"
    @click="() => emit('click')"
  >
    <slot />
  </button>
</template>

<style scoped>
.button {
  border: 1px solid transparent;
  background: var(--info-color);
  border-radius: var(--default-border-radius);
  color: var(--foreground-color);
  padding: 0 12px;
  height: 30px;
  cursor: pointer;

  &[disabled] {
    background-color: var(--disabled-color);
  }

  &.variant-text {
    background: none;
    text-decoration: underline;
    padding: 0;
    font-size: inherit;
    height: max-content;
    color: var(--anchor-color);
  }

  &.size-large {
    font-size: 1.2rem;
    height: 40px;
    padding: 0 16px;
  }

  &.size-small {
    font-size: 0.9rem;
    height: 26px;
  }
}
</style>
