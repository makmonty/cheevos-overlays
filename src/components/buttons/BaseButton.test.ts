import { mount } from '@vue/test-utils';
import { describe, test } from 'vitest';
import BaseButton from './BaseButton.vue';

describe('Component | BaseButton', () => {
  test('should render with defaults', ({ expect }) => {
    const wrapper = mount(BaseButton);
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe('type', () => {
    test('should show a submit button when type is submit', ({ expect }) => {
      const wrapper = mount(BaseButton, {
        props: {
          type: 'submit'
        }
      });

      expect(wrapper.attributes('type')).toBe('submit');
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('variant', () => {
    test('should show a text button when variant is text', ({ expect }) => {
      const wrapper = mount(BaseButton, {
        props: {
          variant: 'text'
        }
      });

      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('size', () => {
    test('should show a large button when size is large', ({ expect }) => {
      const wrapper = mount(BaseButton, {
        props: {
          size: 'large'
        }
      });

      expect(wrapper.html()).toMatchSnapshot();
    });

    test('should show a small button when size is small', ({ expect }) => {
      const wrapper = mount(BaseButton, {
        props: {
          size: 'small'
        }
      });

      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
