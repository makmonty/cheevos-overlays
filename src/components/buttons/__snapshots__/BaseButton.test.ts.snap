// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component | BaseButton > should render with defaults 1`] = `"<button data-v-4e29ae60="" type="button" class="button variant-button size-medium"></button>"`;

exports[`Component | BaseButton > size > should show a large button when size is large 1`] = `"<button data-v-4e29ae60="" type="button" class="button variant-button size-large"></button>"`;

exports[`Component | BaseButton > size > should show a small button when size is small 1`] = `"<button data-v-4e29ae60="" type="button" class="button variant-button size-small"></button>"`;

exports[`Component | BaseButton > type > should show a submit button when type is submit 1`] = `"<button data-v-4e29ae60="" type="submit" class="button variant-button size-medium"></button>"`;

exports[`Component | BaseButton > variant > should show a text button when variant is text 1`] = `"<button data-v-4e29ae60="" type="button" class="button variant-text size-medium"></button>"`;
